grafana/grafana
# Design System: Improve a11y of ConfirmModal

Fixes #66046

My goal was it to make the screenreader read the description of the modal in order to enable users to decide on all the important information. I also tried `aria-labeledby` and `aria-describedby` which didn't lead to the right effect. That's why I decided to use `aria-live="polite"` although it doesn't seem to be the right property according to the use cases in it's documentation.

**How to test it:**
1. Open localhost:3000
2. Navigate to an existing dashboard
3. Add a row by using the _Add_ dropdown in the header
4. Navigate to the _delete_ button in the row header and open the _Delete row_ modal
5. Check what the screen reader tells users when opening the modal and navigating to different buttons
